platform/android/src/style/layers/layer.cpp

```diff
@@ -111,14 +112,31 @@ namespace android {
         }
     };
 
+    jni::Object<mbgl::android::Statement> Layer::getFilter(jni::JNIEnv& env) {
+        using namespace mbgl::style;
+        using namespace mbgl::android::conversion;
+
+        if (layer.is<FillLayer>()) {
```

Comment:
Please replace with visitor like in: https://github.com/mapbox/mapbox-gl-native/pull/9069